### PR TITLE
Add execute block to fix database setup for Snipe-IT

### LINKED ISSUE #5: Database fails to create during Setup/Pre-Flight
Related to [Snipe-IT Issue #5321](https://github.com/snipe/snipe-it/issues/5321)

### PR DESCRIPTION
**Pull request by ehanlon**
- Add `php artisan migrate` execute block to resolve an issue where the database creation fails during Setup/Pre-flight. 
- Set Snipe-IT URL and NGINX server to FQDN if FQDN is set
- Update `['database']['hostname']` to `127.0.0.1` since MySQL doesn't like `localhost`

Fixes #5

### CODE REVIEW

File: attributes/default.rb [hunk 1]
```diff
@@ -3,7 +3,7 @@
 default['snipeit']['doc_root'] = node['snipeit']['path'] + '/public'
 default['snipeit']['locale'] = 'en'
 default['snipeit']['timezone'] = 'US/Pacific'
-default['snipeit']['url'] = 'http://localhost'
+default['snipeit']['url'] = node['fqdn'] ? 'http://' + node['fqdn'] : 'http://' + node['hostname']
```
Comment (americanhanko) [inline attributes/default.rb]: Wonder if Ruby has something that adds this prefix. If not, might be useful to add a method to "webify" a string.

File: attributes/default.rb [hunk 2]
```diff
@@ -20,7 +20,7 @@
 ]
 
 default['snipeit']['database']['domain'] = 'corp.yourcompany.com'
-default['snipeit']['database']['hostname'] = node['hostname']
+default['snipeit']['database']['hostname'] = '127.0.0.1'
```
Comment (americanhanko) [inline attributes/default.rb]: In the past, I've seen that there are differences between using `localhost` and '127.0.0.1` - just curious if you have seen those and why you might've chosen one over the other for the default.

File: recipes/php.rb [hunk 3]
```diff
@@ -34,3 +34,8 @@
   user node['nginx']['user']
   group node['nginx']['group']
 end
+
+execute 'run laravel migrations to setup database' do
```
Comment (americanhanko) [inline recipes/php.rb]: Not familiar with php artisan, but is there a particular reason we need to use the `--force` flag? Without it, could the resource possibly be idempotent?

File: recipes/php.rb [hunk 4]
```diff
@@ -34,3 +34,8 @@
...
+  command ['php', 'artisan', 'migrate', '--force']
```
Comment (americanhanko) [inline recipes/php.rb]: I might consider specifying the full path of `php` or using the `environment` property to set the `PATH` for this command.

File: attributes/default.rb [hunk 5]
```diff
@@ -3,7 +3,7 @@
 default['snipeit']['doc_root'] = node['snipeit']['path'] + '/public'
```
Comment (americanhanko) [inline attributes/default.rb]: I would change this to be:

```ruby
default['snipeit']['doc_root'] = ::File.join node['snipeit']['path'], 'public'
```

File: attributes/default.rb (hunk 2 above)
Comment (ehanlon) [reply attributes/default.rb]: In Laravel, I'd need to specify the Unix socket for `DB_SOCKET` in order to specify `localhost` as the hostname. Snipe-IT supports this. However, this just sacrifices performance over simplicity.

File: recipes/php.rb (hunk 3 above)
Comment (ehanlon) [reply recipes/php.rb]: `php artisan migrate` is intended to be run interactively, and will no-op if there is nothing to migrate.

This is currently a workaround for #5.

**Pull request merged**